warehouse/slave/worker_job.go

```diff
@@ -317,14 +333,29 @@ func (jr *jobRun) path(stagingFileInfo stagingFileInfo) (string, error) {
 
 // loadFilePath generates a unique path for a load file based on the staging file path.
 // Every call to this function will generate a new path even if the same staging file is used.
-func (jr *jobRun) loadFilePath(stagingFileInfo stagingFileInfo) (string, error) {
+// If Warehouse.useDeterministicLoadFileName is true, the load file path will be unique but the file name will be constant.
+func (jr *jobRun) loadFilePath(stagingFileInfo stagingFileInfo, loadFileNamePrefix string) (string, error) {
 	stagingFilePath, err := jr.path(stagingFileInfo)
 	if err != nil {
 		return "", fmt.Errorf("getting staging file path: %w", err)
 	}
 
+	stagingFilePathWithoutExt := strings.TrimSuffix(stagingFilePath, ".json.gz")
+
+	if jr.conf.GetBool("Warehouse.useDeterministicLoadFileName", false) && slices.Contains(warehouseutils.TimeWindowDestinations, jr.job.DestinationType) && len(loadFileNamePrefix) > 0 {
```

Comment:
Why not use deterministic naming for all warehouses ?

Reply:
That's problematic because as you can see, when using deterministic file naming, file manager might not upload the file if object already exists and upload result will have empty location. This is fine for datalakes but for other warehouses we need the location later to copy the loadfile into the warehouse table. If we miss the location, we will miss loading data into warehouse

Reply:
> file manager might not upload the file if object already exists and upload result will have empty location

Once the file name becomes deterministic, we won't have to rely on the output of the upload call. We can populate the file path using the deterministic name.

Reply:
> Question is why do we want to do this for other warehouses?

To make the code simpler (no if checks).